pkg/codegen/nodejs/gen_program.go

```diff
@@ -1108,16 +1108,60 @@ func (g *generator) genComponent(w io.Writer, component *pcl.Component) {
 	g.genTrivia(w, component.Definition.Tokens.GetCloseBrace())
 }
 
-func (g *generator) genConfigVariable(w io.Writer, v *pcl.ConfigVariable) {
-	// TODO(pdg): trivia
+func computeConfigTypeParam(configType model.Type) string {
+	switch pcl.UnwrapOption(configType) {
+	case model.StringType:
+		return "string"
+	case model.NumberType, model.IntType:
+		return "number"
+	case model.BoolType:
+		return "boolean"
+	case model.DynamicType:
+		return "any"
+	default:
+		switch complexType := pcl.UnwrapOption(configType).(type) {
+		case *model.ListType:
+			return fmt.Sprintf("Array<%s>", computeConfigTypeParam(complexType.ElementType))
+		case *model.MapType:
+			return fmt.Sprintf("Record<string, %s>", computeConfigTypeParam(complexType.ElementType))
+		case *model.ObjectType:
+			if len(complexType.Properties) == 0 {
+				return "any"
+			}
 
+			attributeKeys := []string{}
+			for attributeKey := range complexType.Properties {
+				attributeKeys = append(attributeKeys, attributeKey)
+			}
+			// get deterministically sorted attribute keys
+			sort.Strings(attributeKeys)
+
+			objectType := "{"
+			for index, propertyName := range attributeKeys {
+				propertyType := complexType.Properties[propertyName]
+				if index != len(attributeKeys)-1 {
+					objectType += fmt.Sprintf("%s?: %s, ", propertyName, computeConfigTypeParam(propertyType))
+				} else {
+					// last element, do not add a comma
```

Comment:
Could also build a string slice then [strings.Join](https://pkg.go.dev/strings#Join) it.

Reply:
Done ✅